From: Marc Zyngier <marc.zyngier@arm.com>
Date: Mon, 9 Oct 2017 13:17:43 +0100
Subject: irqchip/gic-v3-its: Only send VINVALL to a single ITS
Patch-mainline: v4.15-rc1
Git-commit: 3c1cceeb3d2879d98eda7afd62838fde7b0f920d
References: FATE#323954

Sending VINVALL to all ITSs is completely pointless, as all
we're trying to achieve is to tell the redistributor that
the property table for this VPE should be invalidated.

Let's issue the command on the first valid ITS and be done with it.

Signed-off-by: Marc Zyngier <marc.zyngier@arm.com>
Acked-by: Alexander Graf <agraf@suse.de>
---
 drivers/irqchip/irq-gic-v3-its.c |    5 +++++
 1 file changed, 5 insertions(+)

--- a/drivers/irqchip/irq-gic-v3-its.c
+++ b/drivers/irqchip/irq-gic-v3-its.c
@@ -2549,7 +2549,12 @@
 		if (its_list_map && !vpe->its_vm->vlpi_count[its->list_nr])
 			continue;
 
+		/*
+		 * Sending a VINVALL to a single ITS is enough, as all
+		 * we need is to reach the redistributors.
+		 */
 		its_send_vinvall(its, vpe);
+		return;
 	}
 }
 
